### ISSUE TITLE
route data to same kafka partition based on the value of a primitive

### ISSUE THREAD

**Issue by herc6** [pmacct/pmacct#150]
Is it possible to route the data to the same partition based on a defined primitive ?  In the document I only found that "kafka_partition_key" has to be a string of printable characters. So I think that this feature is not supported  at this moment.

**Comment by paololucente** [comment]
Hi, you are right this is not currently supported (yet) in mainstream code. Do you have a specific primitive in mind? If you see, Pull Requests #101 #102 #103 #104 are precisely about the feature you are asking for and do address some primitives. I intend to merge these soon and make the feature part of final 1.7.0 release. Paolo

**Comment by herc6** [comment]
I would like to use custom defined primitives.

Op 22 aug. 2017 16:40 schreef "Paolo Lucente" <notifications@github.com>:

Hi, you are right this is not currently supported (yet) in mainstream code.
Do you have a specific primitive in mind? If you see, Pull Requests #101
<https://github.com/pmacct/pmacct/pull/101> #102
<https://github.com/pmacct/pmacct/pull/102> #103
<https://github.com/pmacct/pmacct/pull/103> #104
<https://github.com/pmacct/pmacct/pull/104> are precisely about the feature
you are asking for and do address some primitives. I intend to merge these
soon and make the feature part of final 1.7.0 release. Paolo

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/pmacct/pmacct/issues/150#issuecomment-324047520>, or mute
the thread

**Comment by paololucente** [comment]
Then no, that would be one step further ahead.

**Comment by herc6** [comment]
Hello,

Thanks for the answer. Are there plans to support this  ?

**Comment by paololucente** [comment]
Yes. But i can't answer yet what will be your next question: when? For this feature two things need to happen first: 1) merge the relevant PRs and 2) "annotate" custom primitives so that their values can be further used (for ie. tagging, labelling and for this feature). I will mark this issue as feature request and we can leave it open.

**Comment by jrossi** [comment]
Working great !!!! Thank you so much.
